build.sbt

```diff
@@ -55,14 +55,66 @@ lazy val lmRoot = (project in file("."))
     customCommands
   )
 
+val javaOnlySettings: Seq[Setting[_]] = Seq(
+  crossPaths := false,
+  compileOrder := CompileOrder.JavaThenScala,
+  unmanagedSourceDirectories in Compile := Seq((javaSource in Compile).value),
+  crossScalaVersions := Seq(scala211),
+  autoScalaLibrary := false
+)
+
+// defines Java structures used across Scala versions, such as the API structures and relationships extracted by
+//   the analysis compiler phases and passed back to sbt.  The API structures are defined in a simple
+//   format from which Java sources are generated by the datatype generator Projproject
+val lmInterface = project in file("internal") / "util-interface" settings (
+  commonSettings,
+  javaOnlySettings,
+  name := "LM Interface",
```

Comment:
I don't know if the name LM Interface or LM Logging makes sense, since it really isn't about library management.